File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int answer = 0;
    int maxWidth = 0;
    int maxHeight = 0;
    for(auto& size : sizes) {
        int width = max(size[0], size[1]);
        int height = min(size[0], size[1]);

        maxWidth = max(maxWidth, width);
        maxHeight = max(maxHeight, height);
    }    
    
    return maxWidth * maxHeight;
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int w = 0, h = 0;
        for (int[] s : sizes) {
            int a = Math.max(s[0], s[1]);
            int b = Math.min(s[0], s[1]);
            
            w = Math.max(w, a);
            h = Math.max(h, b);
        }
        return w*h;
    }
}